nodebuilder/core/module.go

```diff
@@ -35,13 +35,14 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 			fx.Provide(core.NewBlockFetcher),
 			fxutil.ProvideAs(core.NewExchange, new(libhead.Exchange[*header.ExtendedHeader])),
 			fx.Invoke(fx.Annotate(
-				func(bcast libhead.Broadcaster[*header.ExtendedHeader],
+				func(
```

Comment:
Can you add a comment saying that this is a ConstructFn? This is a place that I have gotten confused before

Reply:
actually why do we `fx.Invoke` this?